lib/trento/software_updates/discovery.ex

```diff
@@ -13,5 +21,87 @@ defmodule Trento.SoftwareUpdates.Discovery do
   def get_relevant_patches(system_id),
     do: adapter().get_relevant_patches(system_id)
 
+  @spec discover_software_updates :: {:ok, {list(), list()}}
+  def discover_software_updates,
+    do:
+      {:ok,
+       Hosts.get_all_hosts()
+       |> Enum.map(&discover_host_software_updates/1)
+       |> Enum.split_with(fn
+         {:ok, _, :discovered, _, _} -> true
+         _ -> false
+       end)}
```

Comment:
Can't you just return `[]` over an erroring discovery? you can keep the logging and you just have to `Enum.flatten` this list instead of playing with tagged tuples.

Reply:
When testing I want to rely on the output of the function, rather than logging :sweat_smile: That's the reason behind it.

We could alternatively return just one list with all the discovery results, successful and errored. However we'd need to do filtering of successful vs errored results in the test to make meaningful assertions. We get that distinction for free by splitting here. I find it convenient.

I simplified a bit the tuple returned in success case from `{:ok, host_id, :discovered, system_id, relevant_patches}` to `{:ok, host_id, system_id, relevant_patches}` :+1: